src/server/pfs/pfs.go

```diff
@@ -146,6 +153,16 @@ type ErrDropWithChildren struct {
 	Commit *pfs.Commit
 }
 
+type ErrPropagateMultipleCommitsInRepo struct {
```

Comment:
This needs documentation. It is pretty complex.

Reply:
Same as the ambiguous error above, this wouldn't be possible if multiple branches in the same repo could not be connected through a provenance relationship.